internal/image/selector.go

```diff
@@ -0,0 +1,160 @@
+package image
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/pkg/errors"
+)
+
+// SelectionStrategy represents a strategy for selecting a single image from a
+// container image repository.
+type SelectionStrategy string
```

Comment:
Another good example of the motivations for this PR. This was formerly `TagSelectionStrategy`.

`image.TagSelectionStrategy` being used to select a "tag" (which is clearly an image) made little sense compared to:

`image.SelectionStrategy`, which properly uses package name for context, doesn't "stutter," and selects an `Image`.